bin/pre-commit-hook.js

```diff
@@ -28,3 +32,28 @@ if ( lintResult.status ) {
 		'repeat the commit command with --no-verify to avoid this check.' );
 	process.exit( 1 );
 }
+
+
+// run prettier for any files in the commit that have @format within their first docblock
+const formatRegex = /@format/;
+const firstDocBlockStart = /\/\*\*/;
+const firstDocBlockEnd = /\*\//;
+
+files
+	.map( file => path.join( __dirname, '../', file ) )
+	.forEach( file => {
+		fs.readFile( file, 'utf8', ( err, data ) => {
+			const firstDocBlockStartIndex = _.get( firstDocBlockStart.exec( data ), 'index' );
+			const firstDocBlockEndIndex = _.get( firstDocBlockEnd.exec( data ), 'index' );
+			const firstDocBlockText =
+				_.isInteger( firstDocBlockStartIndex ) &&
+				_.isInteger( firstDocBlockEndIndex ) &&
+				data.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 2 );
+
+			if ( firstDocBlockText && formatRegex.test( firstDocBlockText ) ) {
+				const formattedData = prettier.format( data, {} );
+				fs.writeFileSync( file, formattedData );
+				execSync( `git add ${ file }` );
+			}
```

Comment:
did you consider just using a basic RegExp and search for the string? we're not really parsing the comments here anyway so I'm not sure we are gaining much by not using one.

```js
const mattchy = /^((?!\/\*\*)(?:\n|.))*?(?:\/\*\*(?:(?!(\/\*\*|\*\/))(\n|.))+)(@format pretty)(?:(?:(\n|.))+\*\/)/;

if ( mattchy.test( data ) ) {
	doItToIt()
}
```

:trollface: 

let's try that again

```js
const mattcher = new RegExp(
	'^' +                                // at start of file
	'(' +                                // look for…
		'(?!\/\*\*)(?:\n|.)' +           // …anything except a comment opener
	')*?' +                              // zero or more times, lazy
	'(?:' +                              // then comes a group
		'\/\*\*' +                       // starting with a comment opener
		'(?:(?!(\/\*\*|\*\/))(\n|.))+' + // and nothing that opens or closes a comment
	')' +                                // this is not optional, we need at least a space
	'(@format pretty)' +                 // then our desired tag
	'(?:(?:(\n|.))+?' +                  // and a bunch of other stuff
	'\*\/)'                              // ending with the closer
);
```

Marginally better, but a program in the RegExp language that gets the job done quickly and succinctly nonetheless, if you ignore the actual RegExp source code.